homeassistant/components/sensor/airvisual.py

```diff
@@ -249,41 +267,48 @@ def device_state_attributes(self):
     def async_update(self):
         """Update the status of the sensor."""
         yield from super().async_update()
-        symbol = self._data.pollution_info.get('main{0}'.format(self._locale))
-        pollution_info = POLLUTANT_MAPPING.get(symbol, {})
-        self._state = pollution_info.get('label')
-        self._unit = pollution_info.get('unit')
-        self._symbol = symbol
+
+        if self._data.pollution_info:
+            symbol = self._data.pollution_info.get(
+                'main{0}'.format(self._locale))
+            pollution_info = POLLUTANT_MAPPING.get(symbol, {})
+            self._state = pollution_info.get('label')
+            self._unit = pollution_info.get('unit')
+            self._symbol = symbol
 
 
 class AirVisualData(object):
     """Define an object to hold sensor data."""
 
-    def __init__(self, client, latitude, longitude, radius):
+    def __init__(self, client, **kwargs):
         """Initialize."""
-        self.city = None
         self._client = client
-        self.country = None
-        self.latitude = latitude
-        self.longitude = longitude
         self.pollution_info = None
-        self.radius = radius
-        self.state = None
+
+        self.city = kwargs.get(CONF_CITY)
+        self.state = kwargs.get(CONF_STATE)
+        self.country = kwargs.get(CONF_COUNTRY)
+
+        self.latitude = kwargs.get(CONF_LATITUDE)
+        self.longitude = kwargs.get(CONF_LONGITUDE)
+        self.radius = kwargs.get(CONF_RADIUS)
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Update with new AirVisual data."""
         import pyairvisual.exceptions as exceptions
 
         try:
-            resp = self._client.nearest_city(self.latitude, self.longitude,
-                                             self.radius).get('data')
+            if self.city and self.state and self.country:
+                resp = self._client.city(self.city, self.state,
+                                         self.country).get('data')
+            else:
+                resp = self._client.nearest_city(self.latitude, self.longitude,
+                                                 self.radius).get('data')
             _LOGGER.debug('New data retrieved: %s', resp)
-
-            self.city = resp.get('city')
-            self.state = resp.get('state')
-            self.country = resp.get('country')
-            self.pollution_info = resp.get('current').get('pollution')
+            self.pollution_info = resp['current']['pollution']
         except exceptions.HTTPError as exc_info:
-            _LOGGER.error('Unable to update sensor data')
+            _LOGGER('Unable to retrieve data from the API')
+            _LOGGER.error("There is likely no data on this location")
             _LOGGER.debug(exc_info)
+            self.pollution_info = None
```

Comment:
You could do `self.pollution_info = {}` so you don't have to do all the checks above.